door_pass/scripts/door_prediction.py

```diff
@@ -80,43 +89,51 @@ def MapCallback(self, msg) :
         self.map_received = True
 
 
+    def fremen_restart_cb(self, msg) :
+        if msg.data:
+            rospy.logwarn("FREMENSERVER restart detected will generate new models now")
+            with self.lock:
+                self.create_models()
+            
+
     """
      BuildCallBack
      
      This Functions is called when the Action Server is called to build the models again
     """
     def build_callback(self, goal):
-        self.cancelled = False
-
-        # print goal
-
-        # set epoch ranges based on goal
-        if goal.start_range.secs > 0:
-            self.range[0] = goal.start_range.secs
-        if goal.end_range.secs > 0:
-            self.range[1] = goal.end_range.secs
-
-        rospy.loginfo('Building model for epoch range: %s' % self.range)
-
-        start_time = time.time()  
-        #self.get_list_of_edges()
-        #elapsed_time = time.time() - start_time
-        #self._feedback.result = "%d edges found in %.3f seconds \nGathering stats ..." %(len(self.eids),elapsed_time)
-        #self._as.publish_feedback(self._feedback)
-        #self.gather_stats()        
-        self.create_models()
-        elapsed_time = time.time() - start_time
-        self._feedback.result = "Finished after %.3f seconds" %elapsed_time 
-        self._as.publish_feedback(self._feedback)       #Publish Feedback
-
-        #rospy.loginfo("Finished after %.3f sechttp://9gag.com/gag/aGR3z60onds" %elapsed_time)
-        
-        if not self.cancelled :     
-            self._result.success = True
-            self._as.set_succeeded(self._result)
-        else:
-            self._result.success = False
-            self._as.set_preempted(self._result)
+        with self.lock:
+            self.cancelled = False
+    
+            # print goal
+    
+            # set epoch ranges based on goal
+            if goal.start_range.secs > 0:
+                self.range[0] = goal.start_range.secs
+            if goal.end_range.secs > 0:
+                self.range[1] = goal.end_range.secs
+    
+            rospy.loginfo('Building model for epoch range: %s' % self.range)
+    
+            start_time = time.time()  
+            #self.get_list_of_edges()
+            #elapsed_time = time.time() - start_time
+            #self._feedback.result = "%d edges found in %.3f seconds \nGathering stats ..." %(len(self.eids),elapsed_time)
+            #self._as.publish_feedback(self._feedback)
+            #self.gather_stats()        
+            self.create_models()
```

Comment:
So it creates models in the `fremen_restart_cb` and here? Seems a bit redundant.